### PR TITLE
Removed tabs logic from sidebar components

### PR DESCRIPTION
**Pull request by mihai-macaneata**
/addresses #105

### CODE REVIEW

**Comment by mihai-macaneata** [comment]
I think it is ready for merge

File: assets/js/components/Allocation-overview.vue [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<template>
```
Comment (xlotlu) [inline assets/js/components/Allocation-overview.vue]: `AllocationOverview.vue` please, like all rest

File: assets/js/components/Allocation-overview.vue [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+<div v-if="hasData">
+    <div class="sidebar-header">
```
Comment (xlotlu) [inline assets/js/components/Allocation-overview.vue]: how about a single div here, and set class from the parent?

File: assets/js/components/Allocation-overview.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      <transition name="fade">
+        <div class="allocation" :key="transitioned">
+          <strong>{{ format(data.allocation) }}</strong>
+          <small>{{ format(data.bilateral_allocation) }} for bilateral relations</small>
+        </div>
+      </transition>
+
+    </div>
+    </div>
+  </div>
```
Comment (xlotlu) [inline assets/js/components/Allocation-overview.vue]: you opened 2 divs but close 3 here

File: assets/js/components/Programmes.vue [hunk 4]
```diff
@@ -1,12 +1,9 @@
 <template>
-  <div class="sidebar-tab-pane programmes"
-       v-bind:class="{
-        'is-selected': selected,
-        'is-loading': loading
-       }">
-    <ul class="sidebar-tab-result-list" v-if="hasData">
+  <section class="sidebar-section-wrapper programmes"
+             role="tabpanel">
```
Comment (xlotlu) [inline assets/js/components/Programmes.vue]: `class="sidebar-section-wrapper"` and `role="tabpanel"` don't belong here, they should be set by the parent

File: assets/js/components/Programmes.vue [hunk 5]
```diff
@@ -164,23 +151,6 @@ export default Vue.extend({
...
-  watch: {
-    selected () { this.loadResults(); },
-  },
-
-  data() {
-    return {
-      loading: false,
-    };
-  },
-
```
Comment (xlotlu) [inline assets/js/components/Programmes.vue]: :+1:

File: assets/js/components/Programmes.vue [hunk 6]
```diff
@@ -35,16 +32,8 @@
...
-    <div class="small muted align-center">
-      &ndash;
-      <button type="button" class="btn-link">show 10 more results</button>
-      |
-      <a href="#search" class="muted">show all</a>
-      &ndash;
-    </div>
-
-  </div>
+  <slot />
```
Comment (xlotlu) [inline assets/js/components/Programmes.vue]: why is this slot here?

File: assets/js/components/Results.vue [hunk 7]
```diff
@@ -21,19 +18,12 @@
...
-    <div class="small muted align-center">
-      &ndash;
-      <button type="button" class="btn-link">show 10 more results</button>
-      |
-      <a href="#search" class="muted">show all</a>
-      &ndash;
-    </div>
-
-  </div>
+  </slot>
```
Comment (xlotlu) [inline assets/js/components/Results.vue]: wut?

File: assets/js/components/Results.vue [hunk 8]
```diff
@@ -1,12 +1,9 @@
 <template>
-  <div class="sidebar-tab-pane results"
-       v-bind:class="{
-        'is-selected': selected,
-        'is-loading': loading
-       }">
-    <ul class="sidebar-tab-result-list" v-if="hasData">
+  <section class="sidebar-section-wrapper results"
+    role="tabpanel">
```
Comment (xlotlu) [inline assets/js/components/Results.vue]: The same, class and role.

Actually, why is this section even needed? Isn't the `ul` container enough?

File: assets/js/components/Sidebar.vue [hunk 9]
```diff
@@ -1,117 +1,35 @@
...
-    </div>
+  <div class="sidebar">
+    <button type="button" id="close-sidebar" class="no-btn"
+                  title="Close results"
+                v-if="isMobileExpanded"
+                v-on:click="mobileCollapse">
+          <span class="icon icon-close"></span>
+  </button>
+    <slot name="Allocation_overview"></slot>
+    <slot name="Tabs"></slot>
```
Comment (xlotlu) [inline assets/js/components/Sidebar.vue]: nope, wrong. why the two slots? if you make the sidebar aware of what's inside it, then you'll end up needing one sidebar component for each scenario. wrong.

File: assets/js/components/index.js [hunk 10]
```diff
@@ -20,6 +22,9 @@ export {
   ProjectsMap,
   Beneficiaries,
   GlobalFilters,
-  Sidebar,
+  Allocation_overview,
```
Comment (xlotlu) [inline assets/js/components/index.js]: seriously, not noticing how everything IsCamelCase and Yours_isnot?

File: assets/js/dummy.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
+const results_data = [
```
Comment (xlotlu) [inline assets/js/dummy.js]: really? 
.....
....
really?? really really?

File: assets/js/root-instances.js [hunk 12]
```diff
@@ -35,6 +36,11 @@ export const Grants = Viz.extend({
     sectors: components.Sectors,
     xmap: components.AllocationMap,
     beneficiaries: components.Beneficiaries,
+    allocation_overview: components.Allocation_overview,
```
Comment (xlotlu) [inline assets/js/root-instances.js]: just "overview" will do

File: templates/grants.html [hunk 13]
```diff
@@ -40,11 +40,22 @@
...
-    <sidebar
-        :initial="dataset"
-    />
-  </div>
+<sidebar>
+  <allocation_overview
+      :initial="dataset"
+      slot="Allocation_overview">
+  </allocation_overview>
+    <tabs slot="Tabs">
```
Comment (xlotlu) [inline templates/grants.html]:
```
spacing
  is
    off
```

**Pull request merged**